pypit/arms.py

```diff
@@ -68,154 +97,160 @@ def ARMS(spectrograph, fitstbl, setup_dict):
         sci_ID = all_sci_ID[sc]
         scidx = np.where((fitstbl['sci_ID'] == sci_ID) & fitstbl['science'])[0][0]
         msgs.info("Reducing file {0:s}, target {1:s}".format(fitstbl['filename'][scidx],
-                                                             fitstbl['target'][scidx])) #slf._target_name))
+                                                             fitstbl['target'][scidx]))
+                                                             #slf._target_name))
         # Loop on Detectors
-        for kk in range(settings.spect['mosaic']['ndet']):
+        for kk in range(_spectrograph.ndet):
             det = kk + 1  # Detectors indexed from 1
-            if settings.argflag['reduce']['detnum'] is not None:
-                if det not in map(int, settings.argflag['reduce']['detnum']):
+            if _par['rdx']['detnum'] is not None:
+                if det not in map(int, _par['rdx']['detnum']):
                     msgs.warn("Skipping detector {:d}".format(det))
                     continue
                 else:
                     msgs.warn("Restricting the reduction to detector {:d}".format(det))
             # Setup
             dnum = settings.get_dnum(det)
             msgs.info("Working on detector {:s}".format(dnum))
+            # TODO: Should this be det or dnum; can we deprecate dnum?
             sci_dict[det] = {}
 
```

Comment:
Seeing lines 86-104 on the left be deleted brings
a real warm and fuzzy feeling to my inner being!